testslide/matchers.py

```diff
@@ -0,0 +1,309 @@
+        return f"{self.a} & {self.b}"
+
+
+class _XorMatcher(_AlreadyChainedMatcher):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def __eq__(self, other):
+        return (self.a == other or self.b != other) and (
+            self.a != other or self.b == other
+        )
+
+    def __repr__(self):
+        return f"{self.a} ^ {self.b}"
+
+
+class _InvMatcher(_AlreadyChainedMatcher):
+    def __init__(self, matcher):
+        self.matcher = matcher
+
+    def __eq__(self, other):
+        return not (self.matcher == other)
+
+    def __repr__(self):
+        return f"! {self.matcher}"
+
+
+class _OrMatcher(_AlreadyChainedMatcher):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def __eq__(self, other):
+        return self.a == other or self.b == other
+
+    def __repr__(self):
+        return f"{self.a} | {self.b}"
+
+
+class _RichComparison(_Matcher):
+    def __init__(self, klass, lt=None, le=None, eq=None, ne=None, ge=None, gt=None):
+        self.klass = klass
+        self.lt = lt
+        self.le = le
+        self.eq = eq
+        self.ne = ne
+        self.ge = ge
+        self.gt = gt
+
+    def __eq__(self, other):
+        if not isinstance(other, self.klass):
+            return False
+        if self.lt is not None and not (other < self.lt):
+            return False
+        if self.le is not None and not (other <= self.le):
+            return False
+        if self.eq is not None and not (other == self.eq):
+            return False
+        if self.ne is not None and not (other != self.ne):
+            return False
+        if self.ge is not None and not (other >= self.ge):
+            return False
+        if self.gt is not None and not (other > self.gt):
+            return False
+        return True
+
+    def __repr__(self):
+        return "<{} 0x{:02X}{}{}{}{}{}{}>".format(
+            type(self).__name__,
+            id(self),
+            f" lt={self.lt}" if self.lt is not None else "",
+            f" le={self.le}" if self.le is not None else "",
+            f" eq={self.eq}" if self.eq is not None else "",
+            f" ne={self.ne}" if self.ne is not None else "",
+            f" ge={self.ge}" if self.ge is not None else "",
+            f" gt={self.gt}" if self.gt is not None else "",
+        )
+
+
+class float_comparison(_RichComparison):
+    """
+    Compares true if other number passes all rich comparison cases given.
+    """
+
+    def __init__(self, lt=None, le=None, eq=None, ne=None, ge=None, gt=None):
+        super().__init__(float, lt=lt, le=le, eq=eq, ne=ne, ge=ge, gt=gt)
+
+
+class int_comparison(_RichComparison):
+    """
+    Compares true if other number passes all rich comparison cases given.
+    """
+
+    def __init__(self, lt=None, le=None, eq=None, ne=None, ge=None, gt=None):
+        super().__init__(int, lt=lt, le=le, eq=eq, ne=ne, ge=ge, gt=gt)
+# Ints
+AnyInt = int_comparison
+
+class ThisInt(int_comparison):
```

Comment:
`ExactInt`? or just `Int`? Or... nothing? Integers already have equality working, I don't think we really need this case.